tests/rules/deprecation.test.ts

```diff
@@ -169,6 +169,38 @@ ruleTester.run('deprecation', rule, {
     getValidTestCase(`
       const component = <Component/>;
     `),
+    getValidTestCase(`
+    class Class {
+      method(param: string): void;
+      /** @deprecated */ method(param: number): void;
+
+      method(param: any): void {}
+    }
+    new Class().method('');
+    const obj = new Class();
+    obj.method('');
+  `),
+  getValidTestCase(`
+    class Class {}
+    interface Interface extends Class {
```

Comment:
This does not make much sense as interfaces should not extend classes.
What is the purpose of this?

Reply:
In a previous version of the TypeScript handbook, this exact technique was mentioned, see https://www.typescriptlang.org/docs/handbook/classes.html#using-a-class-as-an-interface. It's not mentioned anymore in the current version of the handbook, however it is still used in real world.